Webapp/sources/services/solvent_surfer/graph.py

```diff
@@ -124,20 +124,24 @@ def editable_bokeh_graph(
     # colour control points red if applicable
     mapper_control_point = control_point_colours()
 
-    # generate colour mappers depending on colours
-    if colour_name != "CHEM21":
-        if colour_name in exp_data_fields:
-            mapper = yield_colours(colour_name)
+    # define default colour
+    mapper = CHEM21_colours()
 
-        else:
-            mapper = descriptor_colours(colour_name, df)
+    # update colour mapper depending on colour_name
```

Comment:
this is one of those long functions that it would help to break up into smaller functions to make it easier to maintain. 
looks like the comments already define the sections that could be extracted as well

Reply:
same here